### DISCUSSION TITLE
Online files and hard wraps in log files

### DISCUSSION THREAD

**Discussion by ChrisVanBael** [Kittyfisto/Tailviewer#292] [Q&A]
Hi,

I'm investigating TailViewer if it fits my needs.
Is it possible to open files hosted on a webserver? http://intranet.company.com/folder/logfile.log

I can see it seems to make a local copy of the file and open that, but I don't think the tail functionality then still works.
There is also no sort of refresh I think?

And another problem is that my log files are hard wrapped at 120 chars.
Would it be possible to apply a sort of formatter before displaying?
Should this be done with a custom plugin, or does something already exist?

**Comment by Kittyfisto** [comment]
Hi @ChrisVanBael,

> Is it possible to open files hosted on a webserver? http://intranet.company.com/folder/logfile.log

at the moment, Tailviewer *out-of-the-box* can only tail files accessible via the filesystem (be it a local drive or a network share which I use at work mostly).
I have a similar feature request request already (https://github.com/Kittyfisto/Tailviewer/discussions/253) wherein a user wants to Tail Datagrams sent to a specific UDP port which is somewhat similar to your request. I am currently preparing Tailviewer for a 1.0 release which includes support for data sources which are not files (but are accessible via UDP, HTTP, etc..) and I plan on adding UDP data sources right after that. Tailing web-logs could come after that (and build upon the foundation I'm currently building).  

> There is also no sort of refresh I think?

What do you mean by that exactly? Tailviewer does detect **append to**, **deletion** and **truncation** of files, but maybe you mean something different?

> And another problem is that my log files are hard wrapped at 120 chars.
Would it be possible to apply a sort of formatter before displaying?
Should this be done with a custom plugin, or does something already exist?

What would you have this formatter do specifically?  
At the moment, Tailviewer displays files as they are written (e.g. every line in the original log file is displayed in one line). Tailviewer is intelligent enough to find out if multiple lines are the same log entry (most of the time) which helps highlighting and when filtering for specific terms. There's no way as of right now to "wrap" lines or, as I would assume you want to "unbreak lines". Such a thing might be doable by writing a plugin for Tailviewer, but that depends on what should be done about those 120 character breaks.  
I've written various formatters at work which transform log files on a line-by-line basis (such as ensuring that all timestamps are in the  same format) which is both easy to achieve and works very well with the current architecture.
If you tell me a little bit more about what you want to achieve I might be able to give some more detailed answers.
